lib/algolia/client.rb

```diff
@@ -374,16 +377,27 @@ def delete(uri, type = :write)
 
     # This method returns a thread-local array of sessions
     def thread_local_hosts(read)
```

Comment:
I had a hard time following what would happen on the first call where current_index would be `nil`.
```
Thread.current[read ? :algolia_search_host_index : :algolia_host_index] ||= 0
```
would have made everything clearer here, and we could then remove the `.to_i` line 393.